### ISSUE TITLE
Warning when including hap-nodejs into plugin

### ISSUE THREAD

**Issue by ztalbot2000** [homebridge/homebridge#2971]
**Describe The Bug:**

including hap-nodejs in your package.json causes the Red warning:"The plugin "homebridge-cmd4" defines 'homebridge' and/or 'hap-nodejs' in their 'dependencies' section, meaning they carry an additional copy of homebridge and hap-nodejs. This not only wastes disk space, but also can cause major incompatibility issues and thus is considered bad practice. Please inform the developer to update their plugin!"

I'm the developer of Cmd4. I had previously removed this dependancy (Along with homebridge), but want to put hap-nodejs back and the reason is this.
I'm working on my config.schema.json, but specifically with homebridge/plugin-ui-utils.  This plugin has a web server which allows me to configure my schema dynamically at runtime because my Cmd4 plugin tries to support everything which is impossible to do statically.  The server does not get _api where the format can be queried, so I decided to include hap-nodejs and that's where I ran into this issue.  
Is there any way this warning can be removed, or do you know another way around it.

Thanks,
John Talbot

**To Reproduce:**

**Expected behavior:**

[**Logs:**]()

```
Secondary Homebridge Logs here
```

**Homebridge Config:**

```json
Show your homebridge config.json here
```

**Screenshots:**

**Environment:**

* **Node.js Version**: v12.21.0
* **NPM Version**: 7.5.2
* **Homebridge Version**: Initializing HAP-NodeJS v0.9.4...
1.3.4
* **Operating System**: Raspbian / Ubuntu / Debian / Windows / macOS / Docker / other Debian
* **Process Supervisor**: Systemd / init.d / pm2 / launchctl / Docker / hb-service / other / none launchctl

**Comment by ztalbot2000** [comment]
Hi,

  Just to let you know that I noticed Homebridge-gsh clones a lot of information from hap.  I think others may want to include hap in their direct dependencies.
  I did spend the last week trying to get around this issue, but cloning default values or possible values to put into a schéma at runtime, without access to _api is very painful and wastes more disk space than would ever be saved, not to mention the problems of creating out of date values. Cloning for just one accessory is not so bad, but Cmd4 and even gsh which supports so many is impossibly difficult and problematic.

thanks,
John Talbot
